Adding enabled_protocol to azurerm_storage_share

hriaz opened the pull request:
Fixes #861 

Introduces support for Azure File share NFS support.  Requires minimum provider version 2.84

arnaudlh reviewed (changes requested):
thanks for the contribution @hriaz, looks good to me, do you mind adding examples for CI/CD and also visibility? thanks!

hriaz commented:
@arnaudlh Will do thanks.

arnaudlh commented:
Testing and getting: 

╷
│ Error: Missing required argument
│ 
│   on ../modules/storage_account/file_share/file_share.tf line 15, in resource "azurerm_storage_share" "fs":
│   15:       dynamic "access_policy" {
│ 
│ The argument "for_each" is required, but no definition was found.
╵
Terraform plan return code: 1

@hriaz - Can you please have a look if you want this one merged? thanks!

hriaz commented:
> Testing and getting:
> 
> ╷ │ Error: Missing required argument │ │ on ../modules/storage_account/file_share/file_share.tf line 15, in resource "azurerm_storage_share" "fs": │ 15: dynamic "access_policy" { │ │ The argument "for_each" is required, but no definition was found. ╵ Terraform plan return code: 1
> 
> @hriaz - Can you please have a look if you want this one merged? thanks!

Please test. Updated try block and added to CI including example

arnaudlh commented on modules/storage_account/file_share/file_share.tf:
```diff
@@ -6,6 +6,23 @@ resource "azurerm_storage_share" "fs" {
   storage_account_name = var.storage_account_name
   quota                = try(var.settings.quota, null)
   metadata             = try(var.settings.metadata, null)
+  enabled_protocol     = try(var.settings.enabled_protocol, "SMB")
```
```suggestion
  enabled_protocol     = try(var.settings.enabled_protocol, null)
```

arnaudlh commented on examples/storage_accounts/108-file-share-nfs/configuration.tfvars:
```diff
@@ -0,0 +1,69 @@
+      team        = "IT"
+    }
+
+    network = {
```
Comment that section or deployment does not succeed.

arnaudlh reviewed (changes requested):
Almost there :)

arnaudlh reviewed (approved):
LGTM
